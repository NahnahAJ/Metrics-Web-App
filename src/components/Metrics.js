import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { FaAngleDoubleRight } from 'react-icons/fa';
import { getApiData } from '../redux/metrics';
import '../styles/Metrics.modules.css';

const Metrics = () => {
  const dispatch = useDispatch();
  const artData = useSelector((state) => state.apiData.apiData);
  const [searchFilter, setSearchFilter] = useState('');
  const [searchResults, setSearchResults] = useState('');

  const searchItems = (searchValue) => {
    setSearchFilter(searchValue);
    if (searchFilter !== '') {
      const filteredData = artData.filter((item) => Object.values(item)
        .join('')
        .toLowerCase()
        .includes(searchFilter.toLowerCase()));
      setSearchResults(filteredData);
    } else {
      setSearchResults(artData);
    }
  };

  useEffect(() => {
    dispatch(getApiData());
  }, [dispatch]);

  return (
    <>

      <div className="searchBox">
        <input
          className="displays inputCmp"
          type="text"
          placeholder="search here"
          onChange={(e) => searchItems(e.target.value)}
        />
      </div>

      <h2 className="header">Pieces</h2>

      <div className="searchPieces">
        {searchFilter.length > 1 ? (
          searchResults.map((key) => (
            <div key={key.id} className="container displays">
              <span className="svgContainer displays"><Link to={`details/${key.id}`}><FaAngleDoubleRight className="iconSize" /></Link></span>
              <img src={key.imageUrl} alt={key.image_id} />
              <h3>{key.nameEU}</h3>
            </div>
          ))
        ) : (
          <div className="pieces">
            {artData.map((key) => (
              <div key={key.id} className="container displays">
                <div className="svgContainer displays"><Link to={`details/${key.id}`}><FaAngleDoubleRight className="iconSize" /></Link></div>
                <div><img src={key.imageUrl} alt={key.image_id} /></div>
                <span>
                  <h3>{key.nameEU}</h3>
                  <h3 className="numbers">{key.id}</h3>
                </span>
              </div>
            ))}
          </div>
        )}
      </div>
    </>
  );
};

export default Metrics;
