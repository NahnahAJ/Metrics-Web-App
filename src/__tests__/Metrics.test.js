import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Metrics from '../components/Metrics';

const renderFuction = () => {
  render(
    <Provider store={store}>
      <Metrics />
    </Provider>,
  );
};

test('Check if metrics component Renders', () => {
  renderFuction();
  const linkElement = screen.getAllByRole('generic');
  expect(linkElement.length).toBeGreaterThan(1);
});

test('renders the Metrics component snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Metrics />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
