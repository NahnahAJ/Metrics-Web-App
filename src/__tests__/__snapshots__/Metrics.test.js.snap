// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Metrics component snapshot 1`] = `
<DocumentFragment>
  <div
    class="searchBox"
  >
    <input
      class="displays inputCmp"
      placeholder="search here"
      type="text"
    />
  </div>
  <h2
    class="header"
  >
    Pieces
  </h2>
  <div
    class="searchPieces"
  >
    <div
      class="pieces"
    />
  </div>
</DocumentFragment>
`;
